metasynth/distribution/regex/element.py

```diff
@@ -346,10 +358,17 @@ class AnyRegex(BaseRegexClass):
     def __init__(self, min_digit: int, max_digit: int,  # pylint: disable=super-init-not-called
                  extra_char: set[str]=None,
                  frac_used: float=1.0):
-        self.min_digit = min_digit
-        self.max_digit = max_digit
         self.extra_char = set() if extra_char is None else extra_char
-        self.frac_used = frac_used
+        super().__init__(min_digit, max_digit, frac_used)
+
+        # if frac_used > 1 + 1e-6 or frac_used < -1e-6:
+        #     raise ValueError(f"Error initializing RegexElement with fraction used {frac_used}."
+        #                      f"Set the fraction to be used between 0 (never) and 1 (always).")
+        # self.min_digit = min_digit
+        # self.max_digit = max_digit
+        # self.frac_used = frac_used
+
+    def _prepare_regex(self):
```

Comment:
Why is this part commented out?

Also, why the `frac_used < -1e-6` and not just `frac_used < 0.0`? Is there a precision error somewhere?

Reply:
I commented it out this part, because it is done in the `super().__init__` part. I gave it a margin, because of possible rounding errors down the line somewhere. I don't think there's any of that happening in our code, but I was worried it would come from user code somewhere.

I will remove the commented out code.